src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -604,9 +622,18 @@ private static void finish(@CheckForNull final String cookie) {
                 return Executor.of(this);
             }
 
-            @Restricted(NoExternalUse.class) // for Jelly and toString
+            @Exported @Restricted(NoExternalUse.class) // for Jelly and toString
             public String getUrl() {
-                return PlaceholderTask.this.getUrl(); // we hope this has a console.jelly
+                Run<?,?> r = runForDisplay();
+                if (r == null) {
+                    return "";
+                }
+                Jenkins j = Jenkins.getInstance();
+                String base = "";
+                if (j != null) {
+                    base = Util.removeTrailingSlash(j.getRootUrl()) + "/";
+                }
+                return base + r.getUrl();
```

Comment:
No do not touch this method, it is used [here](https://github.com/jenkinsci/workflow-durable-task-step-plugin/blob/8f7dac968c97e2a123befc2419f16e24234ca785/src/main/resources/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution/PlaceholderTask/PlaceholderExecutable/executorCell.jelly#L45) and expects a context-relative string, not a full URL.

For purposes of the REST API you want to introduce a separate method like `getAbsoluteUrl()`; `@Exported` can take an optional `name="url"` to provide the same user-visible effect.